serf/serf.go

```diff
@@ -1098,19 +1102,21 @@ func (s *Serf) handleNodeLeaveIntent(leaveMsg *messageLeave) bool {
 		return false
 	}
 
+	// Always set the lamport time so that if we retransmit below it won't echo
+	// around forever!
```

Comment:
@rboyer we talked about making this comment more extensive:

 - pointing out that the name is a little misleading since the status might not actually change but that the sideeffect or preventing rebroadcasts in the laast case is still needed.
 - Perhaps linking to an issue or PR that described the problems caused by this for future us trying to work out why we moved it here!

I realise this is merged already but perhaps we could update the comment on master separately while context is fresh - might need to be after we have provided a clearer writeup of the issue somewhere public like the Consul linked issue?

Reply:
new PR with this feedback: https://github.com/hashicorp/serf/pull/608